lib/pharos/cluster_manager.rb

```diff
@@ -47,12 +47,17 @@ def validate
     def apply_phases
       apply_phase(Phases::ValidateHost, config.hosts, ssh: true, parallel: true)
       apply_phase(Phases::ConfigureHost, config.hosts, ssh: true, parallel: true)
+      apply_phase(Phases::ConfigureCfssl, config.etcd_hosts, ssh: true, parallel: true)
+      apply_phase(Phases::ConfigureEtcdCa, [config.etcd_host], ssh: true, parallel: false)
+      apply_phase(Phases::ConfigureEtcd, config.etcd_hosts, ssh: true, parallel: true)
 
-      apply_phase(Phases::ConfigureMaster, config.master_hosts, ssh: true, parallel: false)
+      apply_phase(Phases::ConfigureMaster, config.master_hosts, ssh: true, parallel: false, master: config.master_host)
+      apply_phase(Phases::ConfigureMaster, config.master_hosts, ssh: true, parallel: false, master: config.master_host)
       apply_phase(Phases::ConfigureKubelet, config.worker_hosts, ssh: true, parallel: true) # TODO: also run this phase in parallel for the master nodes, if not doing an upgrade?
-      apply_phase(Phases::ConfigureClient, config.master_hosts, ssh: true, parallel: true)
+      apply_phase(Phases::ConfigureClient, [config.master_host], ssh: true, parallel: false)
 
       # master is now configured and can be used
+      apply_phase(Phases::ConfigureClusterInfo, [config.master_host], ssh: true, parallel: false)
```

Comment:
This is a kube API phase, not a node SSH phase? Needs `master: ..` not `ssh: true`

Reply:
Yeah, `ssh` is not required.. but does it need `master`?

Reply:
This somehow feels wrong:
```ruby
apply_phase(Phases::ConfigureClusterInfo, [config.master_host], ssh: false, parallel: false, master: config.master_host)
```

Probably because I need to pass `master_host` twice...

Reply:
Yes, it kind of is, because those phases are not actually host-specific, but run against the cluster. Some phases run multiple instances, some phase classes only ever run one instance... but the meaning of the `master` is consistent across all of those, it's always used to refer to the kube cluster as a whole.